extension/package.json

```diff
@@ -529,6 +529,16 @@
     "configuration": {
       "title": "DVC",
       "properties": {
+        "dvc.focusedProjects": {
```

Comment:
call it currentWorkingProject or just currentProject?

Reply:
the option allows multiple projects to be set

Reply:
what is the semantics of this then?

Reply:
I see from the demo, thanks. Is there a name / term in VS Code when you open multiple projects / workspaces (what is the right term?)?

Reply:
You can open multiple folders inside of a workspace which is known as a multi-root workspace.
https://code.visualstudio.com/docs/editor/multi-root-workspaces.

In terms of the Git extension I cannot find an option to limit/set the repositories to a certain list in the workspace.

Reply:
How do we call this feature in the extension (ability to have multiple roots)? Btw also, how do we determine in the plots, experiments table which one to use?

Reply:
> How do we call this feature in the extension (ability to have multiple roots)?

We do not have a user-facing term for this. They are auto-discovered when the extension is started.

> Btw also, how do we determine in the plots, experiments table which one to use?

We use a quick pick to let the user choose which one they want to open: 

https://user-images.githubusercontent.com/37993418/210023764-ce86651f-e3c0-4de9-b196-9131747ad98a.mov

Reply:
thanks, Matt, a few questions:

- what would happen by default in that repo from the issue? would I see three projects? (one top level and 2 nested?)
- effectively this new focused option limits the number of auto-detected projects? or it also does something else?
- in the current version, if I open experiments and pick the project - does it affect plots automatically? Run commands, etc - or it will keep asking me for every command?

Reply:
> thanks, Matt, a few questions:
> 
> * what would happen by default in that repo from the issue? would I see three projects? (one top level and 2 nested?)

You would see all three projects (they are not related).

> * effectively this new focused option limits the number of auto-detected projects? or it also does something else?

Essentially yes but if for any reason there was one that wasn't found. E,g it is nested more than 1 level deep it could be used to add the full list.

> * in the current version, if I open experiments and pick the project - does it affect plots automatically? Run commands, etc - or it will keep asking me for every command?

If the experiments table is focused for a particular project (i.e is in the active editor) then all top-level experiment-related commands will be run against that project automatically. Anything selected through the UI for that particular project (i.e context menus in the webview or tree) will be automatically run against that project. Same for plots. In terms of opening plots from experiments, if you do it from the experiments webview then it will open the correct project's plots. If not then you will be prompted.

Reply:
> If the experiments table is focused for a particular project (i.e is in the active editor)

To clarify, it means I can open multiple exp tables? 

> Anything selected through the UI

what happens with any other commands (quick pick, play button in params.yaml, etc)?

Reply:
> To clarify, it means I can open multiple exp tables?

Yes

> what happens with any other commands (quick pick, play button in params.yaml, etc)?

If there is a choice between projects and a webview is in the active editor then the webview's project should be automatically chosen. This will work for quick picks. It won't work for the play button in params.yaml because that file is not linked in to the aforementioned behaviour. The extension will also ask you every time for every command if no webviews are open. That's why if you are working with a single project I'd recommend opening it in its own workspace.

Reply:
We can add a quick pick to set this option if we think people will use it.